### PR TITLE
Updated DEVELOPER_GUIDE.md and workflows to reference changes to the supported JDKs.

### PR DESCRIPTION
**Pull request by AWSHurneyt**
Issue #, if available:
https://github.com/opensearch-project/alerting/issues/242

Description of changes:
1. Updated DEVELOPER_GUIDE.md to reference changes to the supported JDKs based on this [doc](https://github.com/opensearch-project/OpenSearch/blob/main/DEVELOPER_GUIDE.md?plain=1).
2. Updated workflows to build using JDK 11.

*CheckList:*
[x] Commits are signed per the DCO using --signoff

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.
For more information on following Developer Certificate of Origin and signing off your commits, please check [here](https://github.com/opensearch-project/alerting/blob/main/CONTRIBUTING.md#developer-certificate-of-origin).

### CODE REVIEW

File: DEVELOPER_GUIDE.md [hunk 1]
```diff
@@ -1,7 +1,7 @@
 - [Developer Guide](#developer-guide)
   - [Forking and Cloning](#forking-and-cloning)
   - [Install Prerequisites](#install-prerequisites)
-    - [JDK 14](#jdk-14)
+    - [JDK 8](#jdk-8)
```
Comment (downsrob) [inline DEVELOPER_GUIDE.md:4]: Should this say JDK 11?

File: DEVELOPER_GUIDE.md [hunk 2]
```diff
@@ -47,7 +65,7 @@ Currently we just put RCF jar in lib as dependency. Plan to publish to Maven and
 8. `./gradlew alertingBwcCluster#mixedClusterTask` launches a cluster with three nodes of bwc version of OpenSearch with alerting and tests backwards compatibility by upgrading one of the nodes with the current version of OpenSearch with alerting, creating a mixed cluster.
 9. `./gradlew alertingBwcCluster#rollingUpgradeClusterTask` launches a cluster with three nodes of bwc version of OpenSearch with alerting and tests backwards compatibility by performing rolling upgrade of each node with the current version of OpenSearch with alerting.
 10. `./gradlew alertingBwcCluster#fullRestartClusterTask` launches a cluster with three nodes of bwc version of OpenSearch with alerting and tests backwards compatibility by performing a full restart on the cluster upgrading all the nodes with the current version of OpenSearch with alerting.
-11. `./gradlew bwcTestSuite` runs all the above bwc tests combined.
+8. `./gradlew bwcTestSuite` runs all the above bwc tests combined.
```
Comment (downsrob) [inline DEVELOPER_GUIDE.md:68]: Should probably be 11. not 8.

File: DEVELOPER_GUIDE.md [hunk 3]
```diff
@@ -18,14 +18,32 @@ Fork this repository on GitHub, and clone locally with `git clone`.
...
 #### JDK 14
 
-OpenSearch components build using Java 14 at a minimum. This means you must have a JDK 14 installed with the environment variable `JAVA_HOME` referencing the path to Java home for your JDK 14 installation, e.g. `JAVA_HOME=/usr/lib/jvm/jdk-14`.
+To run the full suite of tests, download and install [JDK 14](https://jdk.java.net/archive/) and set `JAVA11_HOME`, and `JAVA14_HOME`. They are required by the [backwards compatibility test](./TESTING.md#testing-backwards-compatibility).
 
 ### Setup
 
 1. Clone the repository (see [Forking and Cloning](#forking-and-cloning))
-2. Make sure `JAVA_HOME` is pointing to a Java 14 JDK (see [Install Prerequisites](#install-prerequisites))
+2. Make sure `JAVA_HOME` is pointing to a Java 8 JDK (see [Install Prerequisites](#install-prerequisites))
```
Comment (downsrob) [inline DEVELOPER_GUIDE.md:46]: Should this be 11?

File: DEVELOPER_GUIDE.md (hunk 2 above)
Comment (AWSHurneyt) [reply DEVELOPER_GUIDE.md:68]: Ah, good catch!

File: DEVELOPER_GUIDE.md (hunk 3 above)
Comment (AWSHurneyt) [reply DEVELOPER_GUIDE.md:47]: Changed to 11.

File: DEVELOPER_GUIDE.md (hunk 1 above)
Comment (AWSHurneyt) [reply DEVELOPER_GUIDE.md]: Added sections for 11 and 14.

**Comment by AWSHurneyt** [comment]
Successfully ran `./gradlew build` locally as well.

**Review by praveensameneni — APPROVED** [review]

**Review by downsrob — APPROVED** [review]

**Review by leeyun-amzn — APPROVED** [review]

**Pull request merged**